### PR TITLE
Add support for SHA-256 in openssl-enc format

### PR DESCRIPTION
**Pull request by kholia**
https://blog.malwarebytes.com/cybercrime/2017/07/the-key-to-the-old-petya-has-been-published-by-the-malware-author/ (relevant read).

### CODE REVIEW

File: run/openssl2john.py [hunk 1]
```diff
@@ -65,15 +65,16 @@ def process(filename, plaintext=None, cipher=0, md=0):
                 rdata = rdata.decode("ascii")
             sys.stdout.write("%s:$openssl$%s$%s$8$%s$%s$0$%s$%s$%s\n" %
                              (filename, cipher, md, salt, last_chunk,
-                              len(rdata), rdata, s))
+                              len(rdata) // 2, rdata, s))
```
Comment (magnumripper) [inline run/openssl2john.py]: What's this?

Comment (kholia) [inline run/openssl2john.py]: It's a fix for a bug I introduced recently when I was making openssl2john.py compatible with Python 3.x.

Comment (kholia) [inline run/openssl2john.py]: You said earlier that you don't read Python code! :P

**Comment by kholia** [comment]
There was a stack smashing problem in `BytesToKey` function in this commit earlier. Luckily, our macOS build bot caught it. ASan does not add instrumentation code around external library calls, so it couldn't detect the problem of stack getting smashed by one of the OpenSSL calls in `BytesToKey` function.

File: run/openssl2john.py (hunk 1 above)
Comment (magnumripper) [inline run/openssl2john.py]: Oops, you caught me there 😆  Yeah I usually skim through it more or less quickly. I was mostly wondering about the `// 2` syntax. What's that? A plain division? Why two of them?

Comment (kholia) [inline run/openssl2john.py]: In Python, `//` means integer division.

Comment (magnumripper) [inline run/openssl2john.py]: Ah it's "floor division". I should've googled it in the first place.

**Pull request merged**